tests/test_port_dpb_system.py

```diff
@@ -248,25 +242,380 @@ def test_port_breakout_with_acl(self, dvs):
+        self.dvs_acl.verify_acl_group_num(len(portGroup))
+        self.dvs_vlan.get_and_verify_vlan_member_ids(len(portGroup))
+
+        # Breakout with "-f" option and ensure it succeeds and associations are removed
+        dvs.change_port_breakout_mode(rootPortName, breakoutMode1x, breakoutOption)
+        dpb.verify_port_breakout_mode(dvs, rootPortName, breakoutMode1x)
+        self.dvs_acl.verify_acl_group_num(0)
+        self.dvs_vlan.get_and_verify_vlan_member_ids(0)
+
+        # Cleanup
+
+        # Remove ACL and VLAN tables
+        self.dvs_acl.remove_acl_table(aclTableName)
+        self.dvs_vlan.remove_vlan(vlanID)
+
+        # Verify cleanup
+        self.dvs_acl.verify_acl_table_count(0)
+        self.dvs_vlan.get_and_verify_vlan_ids(0)
+
+        # check ASIC router interface database
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE")
+        intf_entries = tbl.getKeys()
+        # one loopback router interface
+        assert len(intf_entries) == 1
+
+        # Bring up port
+        self.set_admin_status(dvs, "Ethernet8", "up")
+
+        # Create L3 interface
+        self.create_l3_intf("Ethernet8", "");
+
+        # Configure IPv4 address on Ethernet8
+        self.add_ip_address("Ethernet8", "10.0.0.8/31")
+
+        intf_entries = tbl.getKeys()
+        # one loopback router interface and one port based router interface
+        assert len(intf_entries) == 2
+
+        # check ASIC route database
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+        for key in tbl.getKeys():
+            route = json.loads(key)
+            if route["dest"] == "10.0.0.8/31":
+                subnet_found = True
+            if route["dest"] == "10.0.0.8/32":
+                ip2me_found = True
+
+        assert subnet_found and ip2me_found
+
+        # Breakout Ethernet8 WITH "-f" option and ensure cleanup happened
+        dpb.verify_port_breakout_mode(dvs, "Ethernet8", breakoutMode1x)
+        dvs.change_port_breakout_mode("Ethernet8", breakoutMode2x, breakoutOption)
+        dpb.verify_port_breakout_mode(dvs, "Ethernet8", breakoutMode2x)
+
+        # check ASIC router interface database
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE")
+        intf_entries = tbl.getKeys()
+        # one loopback router interface
+        assert len(intf_entries) == 1
+
+        # check ASIC database
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+        for key in tbl.getKeys():
+            route = json.loads(key)
+            if route["dest"] == "10.0.0.8/31":
+                assert False
+            if route["dest"] == "10.0.0.8/32":
+                assert False
+
+        dpb.verify_port_breakout_mode(dvs, "Ethernet8", breakoutMode2x)
+        dvs.change_port_breakout_mode("Ethernet8", breakoutMode1x)
+        dpb.verify_port_breakout_mode(dvs, "Ethernet8", breakoutMode1x)
```

Comment:
better to add return at end of each function. Better of the future programmer of this file.

Reply:
Sure, but could you please explain me the need of return statement? As far I know python methods doesn't require return statements until they return something.

Reply:
just a suggestion, 
When a python function is long, it is easy to lose track. The usual recommendation is to have a single function, not more than a page size. With Python supporting inner functions, it makes more tough. Anyway Pytest has test_<> format, so it is good.